luigi/contrib/rdbms.py

```diff
@@ -27,7 +27,125 @@
 logger = logging.getLogger('luigi-interface')
 
 
-class CopyToTable(luigi.task.MixinNaiveBulkComplete, luigi.Task):
+class _MetadataColumnsMixin(object):
+    """Provide an additinal behavior that adds columns and values to tables
+
+    This mixin is used to provide an additional behavior that allow a task to
+    add generic metadata columns to every table created for both PSQL and
+    Redshift.
+
+    Example:
+
+        This is a use example or how this mixin could come handy and how to use
+        it.
+
+        .. code:: python
+
+            class CommonMetaColumnsBehavior(object):
+                def update_report_execution_date_query(self):
+                    query = 'UPDATE {0} ' \
+                            "SET date_param = DATE '{1}' " \
+                            'WHERE date_param IS NULL'.format(self.table, self.date)
+
+                    return query
+
+                @property
+                def metadata_columns(self):
+                    if self.date:
+                        cols.append(('date_param', 'VARCHAR'))
+
+                    return cols
+
+                @property
+                def metadata_queries(self):
+                    queries = [self.update_created_tz_query()]
+                    if self.date:
+                        queries.append(self.update_report_execution_date_query())
+
+                    return queries
+
+
+            class RedshiftCopyCSVToTableFromS3(CommonMetaColumnsBehavior, redshift.S3CopyToTable):
+                "We have some business override heres that would only add noise to the
+                example, so let's assume that this is only a shell."
+                pass
+
+
+            class UpdateTableA(RedshiftCopyCSVToTableFromS3):
+                date = luigi.Parameter()
+                table = 'tableA'
+
+                def queries():
+                    return [query_content_for('/queries/deduplicate_dupes.sql')]
+
+
+            class UpdateTableB(RedshiftCopyCSVToTableFromS3):
+                date = luigi.Parameter()
+                table = 'tableB'
+    """
+    @property
+    def metadata_columns(self):
+        """Returns the default metadata columns.
+
+        Those columns are columns that we want each tables to have by default.
+        """
+        return []
+
+    @property
+    def metadata_queries(self):
+        return []
+
+    @property
+    def enable_metadata_columns(self):
+        return False
+
+    def _add_metadata_columns(self, connection):
+        cursor = connection.cursor()
+
+        for column in self.metadata_columns:
+            if len(column) == 0:
+                logger.info('Unable to infer column information from column {column} for {table}'.format(column=column, table=self.table))
+                break
```

Comment:
Should this raise an exception?

Reply:
I was thorned with that decision. Raising an exception would halt the pipeline and other tasks from running while breaking out of the function would simply not add the meta_columns like it would be expected.

Breaking would proceed with the execution of the pipeline but may yield potentially unexpected results, while raising would halt everything from running. 

I think that you bring a fair point and I'll implement an exception as this is how it seems that we're managing that type of cases throughout that contrib.

Reply:
I agree that neither solution is the most ideal. But i'm more comfortable being explicit in what must be done to achieve happy path. Anything straying from that to raise an exception.